giraph-core/src/main/java/org/apache/giraph/worker/BspServiceWorker.java

```diff
@@ -217,7 +217,16 @@ public BspServiceWorker(
         getGraphPartitionerFactory().createWorkerGraphPartitioner();
     workerInfo = new WorkerInfo();
     workerServer = new NettyWorkerServer<I, V, E>(conf, this, context,
-        graphTaskManager.createUncaughtExceptionHandler());
+        graphTaskManager.createUncaughtExceptionHandler(
+          (thread, throwable) -> {
+            // If the connection was closed by the client, then we just log
+            // the error, we do not fail the job, since the client will
+            // attempt to reconnect.
+            return throwable.getMessage().startsWith(
+              "Connection reset by peer") ? false : true;
```

Comment:
Is there a better way to determine the exception type? If not, let's make this string as a constant and use everywhere, to avoid breaking this logic.

Reply:
The exception is IOException which is very general, at this point i'd like to distinguish, and the message was the only way i could find.

As opposed to making this a constant, i'll make the whole logic a function.